lib/awesome_spawn.rb

```diff
@@ -116,15 +138,44 @@ def build_command_line(command, params = nil)
   private
 
   def launch(env, command, spawn_options)
-    output, error, status = Open3.capture3(env, command, spawn_options)
+    if command.kind_of?(Array)
+      output, error, status = Open3.pipe_capture3(env, command, spawn_options)
+    else
+      output, error, status = Open3.capture3(env, command, spawn_options)
+    end
     return output, error, status && status.exitstatus
   end
 
-  def parse_command_options(command, options)
+  def normalize_run_opts(*opts)
+    commands, options = if opts.last.kind_of?(Hash)
+                          [opts[0..-2], opts.last]
+                        else
+                          [opts, {}]
+                        end
+
+    if commands.all? { |cmd| cmd.kind_of?(Array) }
```

Comment:
Just noticed I need to make one addition here for the case of when all commands in the pipe are Array's with their own params hash/"tuple array".

```ruby
# Example (doesn't work right with #parse_command_options)
normalize_run_opts [["true", {:params => {:a => nil}], ["false", {:params => [:b, nil]}]]
#=> [["true", {:params => {:a => nil}, "false", {:params => [:b, nil]}], {}]
```